core/src/main/java/bio/terra/pearl/core/service/rule/EnrolleeRuleEvaluator.java

```diff
@@ -1,36 +1,60 @@
-package bio.terra.pearl.core.rule;
+package bio.terra.pearl.core.service.rule;
 
 import bio.terra.pearl.core.antlr.CohortRuleLexer;
 import bio.terra.pearl.core.antlr.CohortRuleParser;
 import lombok.extern.slf4j.Slf4j;
 import org.antlr.v4.runtime.CharStreams;
 import org.antlr.v4.runtime.CommonTokenStream;
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang3.StringUtils;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
-import java.util.Map;
+import java.util.Objects;
 
 @Slf4j
-public class CohortRuleEvaluator {
-    private final Map<String, Object> variableMap;
+public class EnrolleeRuleEvaluator {
+    private final EnrolleeRuleData ruleData;
 
-    public CohortRuleEvaluator(Map<String, Object> variableMap) {
-        this.variableMap = variableMap;
+    public EnrolleeRuleEvaluator(EnrolleeRuleData ruleData) {
+        this.ruleData = ruleData;
     }
 
-    public static boolean evaluateRule(String rule, Map<String, Object> variableMap) throws RuleEvaluationException, RuleParsingException {
-        CohortRuleEvaluator evaluator = new CohortRuleEvaluator(variableMap);
-        return evaluator.evaluateRule(rule);
+    // for evaluations where we need to indicate whether the rule itself is valid
+    public static boolean evaluateRuleChecked(String rule, EnrolleeRuleData ruleData) throws RuleEvaluationException, RuleParsingException {
+        EnrolleeRuleEvaluator evaluator = new EnrolleeRuleEvaluator(ruleData);
+        return evaluator.evaluateRuleChecked(rule);
     }
 
-    public boolean evaluateRule(String rule) throws RuleEvaluationException, RuleParsingException {
+    // Evaluates the rule. Returns false if the rule cannot be parsed, or if an error occurred accessing the ruleData
+    public static boolean evaluateRule(String rule, EnrolleeRuleData ruleData) {
+        try {
+            return evaluateRuleChecked(rule, ruleData);
+        } catch (RuleEvaluationException | RuleParsingException e) {
+            log.warn("Error evaluating rule [ {} ]: studyEnvironment {}, enrollee {}: ".formatted(rule,
+                    ruleData.getEnrollee().getStudyEnvironmentId(),
+                    ruleData.getEnrollee().getShortcode()), e);
+            return false;
+        }
+    }
+
+
+    public boolean evaluateRuleChecked(String rule) throws RuleEvaluationException, RuleParsingException {
+        if (StringUtils.isBlank(rule)) {
+            return applyDefaultRule();
+        }
         CohortRuleLexer lexer = new CohortRuleLexer(CharStreams.fromString(rule));
         CommonTokenStream tokens = new CommonTokenStream(lexer);
         CohortRuleParser parser = new CohortRuleParser(tokens);
         CohortRuleParser.ExprContext exp = parser.expr();
         return evaluateExpression(exp);
     }
 
+    // the default rule is that the action is applied if the enrollee is a subject
+    protected boolean applyDefaultRule() {
+        return ruleData.getEnrollee().isSubject();
+    }
```

Comment:
Is `isSubject()` something we always want to be a part of the rule unless they specify otherwise? Or maybe we just need to be careful to add documentation around the fact that proxies are included by default which is something they might not want.

Reply:
that's a really good question -- essentially, is this a default rule, or is it also a default condition that gets added to every rule unless the rule explicitly mentions subject.  will put some thought into that -- we don't want A-T to add a rule "sexAtBirth = 'M'" to a survey thinking that will limit it to male subjects, and then suddenly the survey gets assigned to proxies too.